### ISSUE TITLE
Document Minecart entity properties

### ISSUE THREAD

**Issue by madmaxoft** [cuberite/cuberite#3343]
Currently the `cMinecart` and its descendants have lots of properties that are undocumented and have cryptic names, such as `LastDamage()` (also: should be `GetLastDamage()` and `GetBlockHeight()`. 

Doxy-comments should be added to these to indicate what they represent, and optionally who updates them and who uses them.

**Comment by bibo38** [comment]
> and optionally who updates them and who uses them.

No! Don't do that. This should be the job of an IDE, because a comment is likely to break, when somebody else uses the functions in some other place.

> lots of properties that are undocumented and have cryptic names

In my opinion are most of the undocumented properties of `cMinecart` well named. I would only call `LastDamage`(getter not visible + unclear, if last damage tick or last damage amount is meant), `GetContent`(chest content? contained player? contained item?) and `GetPayload`(same as `GetContent`) cryptic.

Furthermore I would rename these methods instead of commenting them, because this improves the readability of the code using these properties. Also Getter/Setter methods should only need a comment, when they are doing something unexpected or complicated. Otherwise these comments just repeat the name of the method and the type of the return value.

Here are my renaming ideas:

- `LastDamage` -> `GetLastDamageAmount`
- `GetContent` -> `GetContainedItem`
- `GetPayload` -> `GetType` or `GetMinecartType`

P.S.: @madmaxoft Why did you mentioned `GetBlockHeight`?

**Comment by madmaxoft** [comment]
"who updates them" is not meant to be the function that calls the update, but the game event that triggers the update. Take `LastDamage` for example - is it the damage that has been applied in this tick (does it get reset upon the end of the tick), or the last tick (reset when?) or the last time the minecart was hit by something?

As for `GetBlockHeight` - what is a block height in relation to minecarts? Is it the height of the minecart above blocks (tracks)? Is it an offset of the block in the minecart visual from the minecart's pivot point? Or is it the height (size) of that block?

**Comment by madmaxoft** [comment]
`GetContainedItem` would imply an item in the minecart (as in: an item in the minecart's chest), while I assume you meant the potential block in the minecart's visual (chest, furnace, commandblock, tnt). Right?

**Comment by bibo38** [comment]
Kinda right. `GetContainedBlock` would be a better name. I just looked at the cItem return type and named the function like that.

But this function exists in the RidableMinecart and should return the custom block(not the block of the minecart type) in the minecart(grass, soul sand, etc. ). As described [here](http://minecraft.gamepedia.com/Minecart#History):
> Minecarts can now be edited with a third party program to show any block inside of it (it will not take on the characteristics of this block), as well as make it take on the characteristics of any cart.

**Comment by Pokechu22** [comment]
All types of minecarts can contain custom blocks, not just rideable ones.  If there is a custom block set, then the contained block is replaced, otherwise it uses the default (air for rideable, chest for chest, furnace/lit furnace for furnace minecart, ect).

The offset defaults to 6 for all types of minecart except for hopper minecarts, where it is 1, and chest minecarts, where it is 8 (however, cuberite doesn't need to send those values if it's not a custom contained block).  The unit is 16ths of a block.

I recommend changing `GetContent` to 2 methods:  `GetContainedBlock` (returning `BLOCKTYPE`) and `GetContainedBlockMeta` (returning `NIBBLETYPE`), and moving it up to `cMinecart` from `cMinecartRideable`.  The constructor should also be changed to _optionally_ take contained items, with the default height being 6.  Also, setter methods for contained blocks _are_ needed - mainly because it's useful for plugins, but also furnace minecarts provide a vanilla case where the block changes.
